frontend/packages/data-portal/e2e/pagination/testPagination.ts

```diff
@@ -0,0 +1,103 @@
+import { ApolloClient, NormalizedCacheObject } from '@apollo/client'
+import { expect, Page, test } from '@playwright/test'
+import { getApolloClient } from 'e2e/apollo'
+import { E2E_CONFIG } from 'e2e/constants'
+import { TableValidator } from 'e2e/filters/types'
+import { goTo } from 'e2e/filters/utils'
+
+import { TestIds } from 'app/constants/testIds'
+
+function getPagination(page: Page) {
+  return page.getByTestId(TestIds.Pagination)
+}
+
+export function testPagination({
+  filterParams,
+  getMaxPages,
+  rowLoadedSelector,
+  url,
+  validateTable,
+}: {
+  filterParams: URLSearchParams
+  getMaxPages(
+    client: ApolloClient<NormalizedCacheObject>,
+    filterParams?: URLSearchParams,
+    pageNumber?: number,
+  ): Promise<number>
+  rowLoadedSelector: string
+  url: string
+  validateTable: TableValidator
+}) {
+  test.describe(url.replace(E2E_CONFIG.url, ''), () => {
+    let client = getApolloClient()
+
+    test.beforeEach(() => {
+      client = getApolloClient()
+    })
+
+    test('should show correct max page count', async ({ page }) => {
+      await goTo(page, url)
+
+      const node = getPagination(page)
+      const maxPages = await getMaxPages(client)
+      const pageNodes = node.locator('li')
+
+      const count = await pageNodes.count()
+      await expect(pageNodes.nth(count - 2)).toContainText(`${maxPages}`)
```

Comment:
also just curious, what does the `count - 2` logic come from? Is it part of the SDS pagination spec or a business logic? 

Not sure if it's worth documenting that magic number 😄